### ISSUE TITLE
Support mypy --python-executable

### ISSUE THREAD

**Issue by tyrion** [typeddjango/django-stubs#125]
I was successfully using django-stubs until recently. It appears that recently you started importing `DJANGO_SETTINGS_MODULE` in the code of the plugin. This breaks for everyone that is running mypy from an other python interpreter than the one where the project is installed.

Many people prefer to install mypy (and other tools/linters) only once in a dedicated virtualenv instead of installing it in every project and then pointing it to the correct python version using `--python-executable`. It would be great if you could support this scenario again.

I think you may be able to locate the settings module by using some mypy internal lookup methods instead of importing it directly. In case that is not possible, or meanwhile you figure it out, it would be great to have the plugin work even if it cannot load `DJANGO_SETTINGS_MODULE`, maybe with limited functionality.

Thank you very much :)

**Comment by proofit404** [comment]
Hi, thanks for the report.

The plugin isn't only import settings module. It also runs `django.setup`. As you can imagine, this plugin relies on proper environment installation in its heart.

With this approach, we exchange the ability to implement little usability features for useful type information. All this with as little maintenance effort as possible.

Given that serious project should run mypy as part of the CI pipeline, which requires the proper environment installation, I suggest marking this issue as won't fix.

Hope my explanation makes sense.

Best regards,
Artem.

**Comment by tyrion** [comment]
Hello @proofit404 ,

Thank you for the reply.
Your explanation makes perfect sense. Feel free to close the issue if you think it is not currently possible to solve it, or if doing so would be too much effort compared to the benefits it would bring.

Cheers
